pandas/_version.py

```diff
@@ -48,8 +49,8 @@ class NotThisMethod(Exception):
     pass
 
 
-LONG_VERSION_PY = {}
-HANDLERS = {}
+LONG_VERSION_PY = {}  # type: Dict
+HANDLERS = {}  # type: Dict[str, Dict]
```

Comment:
It looks like this is generated by `register_vcs_handler` so you should annotate that function to match the appropriate signature here

Reply:
you mean something like this?
    def register_vcs_handler(
        vcs,  # type: Something
        method
    ):

Reply:
Yes but use Py3 syntax

Reply:
This?

    def register_vcs_handler(vcs: str, method: str) -> Callable:

But this will not work in `3.5`

Reply:
I think that annotation is fine within the signature. Just doesn't work for local / class variables until Py36. We already use elsewhere

Reply:
OK, I will use the new syntax. I don't have python 3.5 on my system so I can't check, I will let travis do the checking.

Reply:
@WillAyd I have this for now:

    def register_vcs_handler(vcs: str, method: str) -> Callable:  # decorator

I can specify the arg type and return type of the returned method from `register_vcs_handler` too, but since it's a decorator there are a couple of methods nested and it will get a bit confusing. What do you want me to do?

    def register_vcs_handler(vcs: str, method: str) -> Callable[[Callable], Callable]:  # might get even more nested

Reply:
Yea it looks like decorator annotations are a little tricky. Here's a relevant issue:

https://github.com/python/mypy/issues/3157

Would want to give that a look to see if there's a solution therein

Reply:
I see that mypy doesn't has a perfect way of typing decorators yet. I would suggest we should leave the decorator for now. We can type it later when mypy has better support for decorators.
I have typed them anyway in a way which I thought looked the best out of all options. Take a look, and let me know what seems like the best option

Reply:
I edited the last comment a lot, so don't trust the text of email github sent you.